bitcoin/src/blockdata/transaction.rs

```diff
@@ -497,24 +512,36 @@ impl TxOut {
     pub const NULL: Self =
         TxOut { value: Amount::from_sat(0xffffffffffffffff), script_pubkey: ScriptBuf::new() };
 
-    /// The weight of the txout in witness units
+    /// The weight of this output.
+    ///
+    /// Keep in mind that when adding a [`TxOut`] to a [`Transaction`] the total weight of the
+    /// transaction might increase more than `TxOut::weight`. This happens when the new output added
+    /// causes the output length `VarInt` to increase its encoding length.
     ///
-    /// Keep in mind that when adding a TxOut to a transaction, the total weight of the transaction
-    /// might increase more than `TxOut::weight`. This happens when the new output added causes
-    /// the output length `VarInt` to increase its encoding length.
+    /// # Panics
+    ///
+    /// If this outputs scriptPubkey size exceeds approximately 2^62.
     pub fn weight(&self) -> Weight {
-        let script_len = self.script_pubkey.len();
-        // In vbytes:
-        // value (8) + script varint len + script push
-        Weight::from_non_witness_data_size((8 + VarInt::from(script_len).len() + script_len) as u64)
+        // Size is equivalent to virtual size since all bytes of a TxOut are non-witness bytes.
+        match Weight::from_vb(self.size() as u64) {
+            Some(weight) => weight,
+            // This should not happen under normal conditions, but in case someone is doing
+            // something malicious return max so as not to silently overflow.
+            None => Weight::MAX,
```

Comment:
It doesn't seem any better to return MAX silently instead of overflow silently.  Could None be returned instead of Max if the bounds are exceeded so that the consumer knows.

Reply:
In my opinion this code shows that the `Weight::from_vb` API is wrong and that it should do saturating multiplication, please see https://github.com/rust-bitcoin/rust-bitcoin/issues/2086.